src/jit/lower.h

```diff
@@ -48,34 +48,67 @@ class Lowering : public Phase
     static void getCastDescription(GenTreePtr treeNode, CastInfo* castInfo);
 
     // LowerRange handles new code that is introduced by or after Lowering.
-    void LowerRange(LIR::Range&& range)
+    void LowerRange(LIR::ReadOnlyRange&& range)
     {
         for (GenTree* newNode : range)
         {
             LowerNode(newNode);
         }
     }
+    void LowerRange(GenTree* firstNode, GenTree* lastNode)
+    {
+        LowerRange(LIR::ReadOnlyRange(firstNode, lastNode));
+    }
 
 private:
+    // ContainCheckRange handles new code that is introduced by or after Lowering,
+    // and that is known to be already in Lowered form.
+    void ContainCheckRange(LIR::Range&& range)
+    {
+        for (GenTree* newNode : range)
+        {
+            ContainCheckNode(newNode);
+        }
+    }
+    void ContainCheckRange(GenTree* firstNode, GenTree* lastNode)
+    {
+        for (GenTree* node = firstNode; node != lastNode->gtNext; node = node->gtNext)
+        {
+            ContainCheckNode(node);
+        }
+    }
+
+    void InsertTreeBeforeAndContainCheck(GenTree* insertionPoint, GenTree* tree)
+    {
+        LIR::Range range = LIR::SeqTree(comp, tree);
+        BlockRange().InsertBefore(insertionPoint, std::move(range));
+        ContainCheckRange(std::move(range));
```

Comment:
I don't think that this will work, as `std::move` invalidates its source operand. Flipping the order of the contain check and insert should be fine:

```
LIR::Range range = LIR::SeqTree(comp, tree);
ContainCheckRange(range);
BlockRange().InsertBefore(insertionPoint, std::move(range));
```